crates/rpc/rpc/src/eth/api/sign.rs

```diff
@@ -0,0 +1,38 @@
+//! Contains RPC handler implementations specific to sign endpoints
+use std::ops::Deref;
+
+use crate::{
+    eth::{
+        error::{EthApiError, EthResult},
+        signer::EthSigner,
+    },
+    EthApi,
+};
+use ethers_core::types::transaction::eip712::TypedData;
+use reth_primitives::{Address, Bytes, Signature};
+use serde_json::Value;
+impl<Client, Pool, Network> EthApi<Client, Pool, Network> {
+    pub(crate) async fn sign(&self, account: Address, message: Bytes) -> EthResult<Signature> {
+        let signer = self.find_signer(&account).ok_or(EthApiError::UnknownAccount)?;
+        let signature =
+            signer.sign(account, &message).await.map_err(|_err| EthApiError::UnknownAccount)?; // Maybe this should be another error?
```

Comment:
imo, it would be better to use an opaque signer error variant here (Box< dyn Error>)